source/common/upstream/health_checker_impl.h

```diff
@@ -173,12 +172,24 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
 
     // Http::StreamCallbacks
     void onResetStream(Http::StreamResetReason reason) override;
-
-    // Network::ConnectionCallbacks
-    void onEvent(uint32_t events) override;
     void onAboveWriteBufferHighWatermark() override {}
     void onBelowWriteBufferLowWatermark() override {}
 
+    void onEvent(uint32_t events);
+
+    class ConnectionCallbackImpl : public Network::ConnectionCallbacks {
+    public:
+      ConnectionCallbackImpl(HttpActiveHealthCheckSession& parent) : parent_(parent) {}
+      // Network::ConnectionCallbacks
+      void onEvent(uint32_t events) override { parent_.onEvent(events); }
+      void onAboveWriteBufferHighWatermark() override {}
```

Comment:
"coverage annotations to disregard" don't currently work until we switch to lcov. You can bug @htuch about that. 😉 

In this case though, I'm not sure we should disregard as these functions will likely get called. I feel like unit tests on filters, etc. should probably call the functions when they can just to future proof anything strange happening, but not a big deal.